### PR TITLE
Added unittests for the Main Server

### PR DESCRIPTION
**Pull request by Ranray**
Added unittests for the following classes:

- RMI Stubs
- PasswordUtilities

### CODE REVIEW

**Review by bras1223 — APPROVED** [review]

File: Soccar [Main-Server]/test/nl/soccar/mainserver/rmi/IClientAuthenticatedTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * @author PTS34A
+ */
+public class IClientAuthenticatedTest {
+
+    // Declaration of test object.
+    private IClientAuthenticated clientAuthenticated;
+
+    /**
+     * Instantiation of test objects.
```
Comment (Laugslander) [inline Soccar [Main-Server]/test/nl/soccar/mainserver/rmi/IClientAuthenticatedTest.java]: `// Instantiation of test object.`

File: Soccar [Main-Server]/test/nl/soccar/mainserver/rmi/IClientAuthenticatedTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    /**
+     * Tests the getStatistics Method.
+     *
+     * @throws RemoteException Thrown when a communication error occurs during
+     * the remote call of this method.
+     */
+    @Test
+    public void getStatisticsTest() throws RemoteException {
+        Assert.assertNotNull(clientAuthenticated.getStatistics("username"));
```
Comment (Laugslander) [inline Soccar [Main-Server]/test/nl/soccar/mainserver/rmi/IClientAuthenticatedTest.java]: `Assert.` kan weg.

File: Soccar [Main-Server]/test/nl/soccar/mainserver/rmi/IClientUnauthenticatedTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * @author PTS34A
+ */
+public class IClientUnauthenticatedTest {
+
+    // Declaration of test object.
+    private IClientUnauthenticated clientUnauthenticated;
+
+    /**
+     * Instantiation of test objects.
```
Comment (Laugslander) [inline Soccar [Main-Server]/test/nl/soccar/mainserver/rmi/IClientUnauthenticatedTest.java]: `// Instantiation of test object.`

File: Soccar [Main-Server]/test/nl/soccar/mainserver/rmi/IClientUnauthenticatedTest.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+     * @throws RemoteException Thrown when a communication error occurs during
+     * the remote call of this method.
+     */
+    @Test
+    public void checkIfExistsTest() throws RemoteException {
+        assertTrue(clientUnauthenticated.checkIfExists("username"));
+    }
+
+    /**
+     * Tests the checkPassword method.
```
Comment (Laugslander) [inline Soccar [Main-Server]/test/nl/soccar/mainserver/rmi/IClientUnauthenticatedTest.java]: `Method`

File: Soccar [Main-Server]/test/nl/soccar/mainserver/rmi/IMainServerForGameServerTest.java [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+ * 
+ * @author PTS34A
+ */
+public class IMainServerForGameServerTest {
+
+    // Declaration of test object.
+    private IMainServerForGameServer mainServerForGameServer;
+
+    /**
+     * Instantiation of test objects.
```
Comment (Laugslander) [inline Soccar [Main-Server]/test/nl/soccar/mainserver/rmi/IMainServerForGameServerTest.java:22]: `// Instantiation of test object.`

File: Soccar [Main-Server]/test/nl/soccar/mainserver/util/test/PasswordUtilitiesTest.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    private static final int DEFAULT_SALT_LENGTH = 16;
+    private static final int HASHED_PASSWORD_LENGTH = 64;
+    
+    // Declaration of test object.
+    private final String password = "Welkom123";
+
+    /**
+     * Tests the generalteSalt Method.
+     *
```
Comment (Laugslander) [inline Soccar [Main-Server]/test/nl/soccar/mainserver/util/test/PasswordUtilitiesTest.java]: Enter verwijderen.

File: Soccar [Main-Server]/test/nl/soccar/mainserver/util/test/PasswordUtilitiesTest.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        byte[] salt = PasswordUtilities.generateSalt();
+        // Checks if the salt is not null.
+        assertNotNull(salt);
+        // Checks if the salt is 16 long.
+        assertEquals(DEFAULT_SALT_LENGTH, salt.length);
+    }
+    
+    /**
+     * Tests the addSalt Method.
+     *
```
Comment (Laugslander) [inline Soccar [Main-Server]/test/nl/soccar/mainserver/util/test/PasswordUtilitiesTest.java]: Enter verwijderen.

File: Soccar [Main-Server]/test/nl/soccar/mainserver/util/test/PasswordUtilitiesTest.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    @Test
+    public void hashingTest() {
+        byte[] salt = PasswordUtilities.generateSalt();
+        // Checks if the salt is not null.
+        assertNotNull(salt);
+        // Checks if the salt is 16 bytes long.
+        assertEquals(DEFAULT_SALT_LENGTH, salt.length);
+        
+        byte[] hashedPassword = PasswordUtilities.addSalt(password.getBytes(), salt);
+        //Checks if the hashedPassword is not null
```
Comment (Laugslander) [inline Soccar [Main-Server]/test/nl/soccar/mainserver/util/test/PasswordUtilitiesTest.java:45]: `//Checks if the hashedPassword is not null.`

**Review by Laugslander — CHANGES_REQUESTED** [review]

**Review by Laugslander — APPROVED** [review]

**Pull request merged**